lib/mqtt/mqtt.cc

```diff
@@ -575,8 +582,14 @@ SObj mqtt_connect(Timeout                    *t,
 	  reinterpret_cast<MQTTPubAckInfo_t *>(&context->variableLengthData);
 	MQTTPubAckInfo_t *outgoingPublishes =
 	  incomingPublishes + incomingPublishCount;
-	uint8_t *networkBuffer = reinterpret_cast<uint8_t *>(outgoingPublishes) +
-	                         sizeof(MQTTPubAckInfo_t) * outgoingPublishCount;
+	uint8_t *networkBuffer =
+	  static_cast<uint8_t *>(heap_allocate(t, allocator, networkBufferSize));
+
+	if (networkBuffer == nullptr)
```

Comment:
Since we are currently fixing this... it would be nice to check the tag bit instead of `nullptr` here?